zinc/src/main/scala/sbt/internal/inc/MixedAnalyzingCompiler.scala

```diff
@@ -91,26 +97,42 @@ final class MixedAnalyzingCompiler(
           val incToolOptions =
             IncToolOptions.of(
               Optional.of(classfileManager),
-              incOptions.useCustomizedFileManager()
+              config.incOptions.useCustomizedFileManager()
             )
-          val joptions = options.javacOptions().toArray[String]
-          javac.compile(
-            javaSrcs,
-            joptions,
-            output,
-            callback,
-            incToolOptions,
-            reporter,
-            log,
-            progress
-          )
+          val joptions = config.currentSetup.options.javacOptions
+
+          JarUtils.getOutputJar(output) match {
```

Comment:
I remember you mentioned we don't support forked javac compilation. Can we not do a best effort to support it?

Reply:
Maybe I explained something wrong or you misunderstood, but we actually do support it. 

One thing changed in javac is closing the custom file manager. This is not relevant for forked javac as it just exists. The other thing is `-XDuseOptimizedZips=false`. This flag is just passed to javac. If it is forked, it just works. If it is local it doesn't because we create custom file manager and the logic of applying options there is stupid, hence I needed the hack with classloaders and reflections. 

But as I tested it, it works with both forked and local.

Reply:
Then no problem :smile: I remembered having read that it wasn't working with forked compilation but it was probably my misremembering.